### ISSUE TITLE
IonicStorage 1.1.6 - Opening database is delayed

### ISSUE THREAD

**Issue by bimal72** [driftyco/ionic-storage#20]
Hello,
With reference to Ionic Storage - QuotaExceededError - error - #10
I have now used @ionic/Storage 1.1.6 withcordova-sqlite-storage - 1.4.8
 In my platform.ready() code

I have used 

```
storage.get('introShown').then((res) =>{
});
```

it gives me res = null;

The reason is database is opening delayed after my call to get.

```
getting data for.... env
main.js:29 DEVICE READY FIRED AFTER 697 ms
main.js:30 getting data for.... introShown
main.js:29 Ionic Storage driver: cordovaSQLiteDriver
main.js:30 setting data for.... introShown  value... true
SQLitePlugin.js:175 OPEN database: _ionicstorage
SQLitePlugin.js:106 new transaction is waiting for open operation
SQLitePlugin.js:179 OPEN database: _ionicstorage - OK
SQLitePlugin.js:80 DB opened: _ionicstorage
```

**Comment by mirkonasato** [comment]
There is a race condition. Especially when using the SQLite driver it's possible that you call `get` and you get a result from e.g. IndexedDB because the SQLite driver hasn't been initialised yet. Then you call `get` two seconds later and you get a result from SQLite.

As a workaround waiting for [Platform.ready](http://ionicframework.com/docs/v2/api/platform/Platform/#ready) before reading from Storage seems to work, but a better solution would be nice.

**Comment by mirkonasato** [comment]
Here's a [possible fix](https://github.com/mirkonasato/ionic-storage/commit/07acbf94e534c3fb5b249d2ea6dddd21ad9c2a83). Will test it a bit more tomorrow.

**Comment by mnasyrov** [comment]
@mirkonasato Added some thoughts about the fix in your commit.

**Comment by bimal72** [comment]
Thanks all, 
Let me know the final solutions.
As of now I did workaround - keep setTimeout for 100 ms and it is working in all my test devices, Android and iOS .. though it is not right way

**Comment by wwallace** [comment]
It's interesting that setTimeout works for so many people. I am seeing the issue consistently on the app's first run regardless of when I call storage.get() if I have injected Storage into the constructor of app.component.ts or the root page constructor.

On the app's first run, console log reveals that the storage driver is async but if you refresh, it is webSQL. I'm still debugging what the behavior is in native iOS and Android.

**Comment by mirkonasato** [comment]
Given that Ionic Storage is basically just a wrapper around [localForage](https://localforage.github.io/localForage/), I would consider using localForage directly. At least that way you have more control over its initialisation process.

**Comment by mlynch** [comment]
Fixed with #21
